debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/FieldSelector.java

```diff
@@ -0,0 +1,488 @@
+         *
+         * @param doc the document to modify fields
+         */
+        private void modifyFieldsWithDotNotation(Document doc) {
+            // document can contain null value by key
+            if (doc.containsKey(field)) {
+                modifyFieldWithDotNotation(doc, field);
+                return;
+            }
+            List<Entry> deferred = null;
+            Iterator<Map.Entry<String, Object>> it = doc.entrySet().iterator();
+            while (it.hasNext()) {
+                Map.Entry<String, Object> entry = it.next();
+                String[] originKeyNodes = entry.getKey().split("\\.");
+                String[] keyNodes = excludeNumericItems(originKeyNodes);
+                if (keyNodes.length > fieldNodes.length) {
+                    // field is a prefix of key, e.g. field is 'a' and key is 'a.b'
+                    if (startsWith(fieldNodes, keyNodes)) {
+                        deferred = modifyFieldDeferred(deferred, originKeyNodes, entry.getValue());
+                        it.remove();
+                    }
+                } else if (keyNodes.length < fieldNodes.length) {
+                    // key is a prefix of field, e.g. field is 'a.b' and key is 'a'
+                    if (startsWith(keyNodes, fieldNodes)) {
+                        Object value = entry.getValue();
+                        if (value instanceof Document) {
+                            modifyFields((Document) value, fieldNodes, keyNodes.length);
+                        } else {
+                            modifyFields(value, fieldNodes, keyNodes.length);
+                        }
+                        break;
+                    }
+                } else {
+                    // key is equal to field, e.g. field is 'a' and key is 'a'
+                    if (Arrays.equals(keyNodes, fieldNodes)) {
+                        deferred = modifyFieldDeferred(deferred, originKeyNodes, entry.getValue());
+                        it.remove();
+                    }
+                }
+            }
+            if (deferred != null) {
+                deferred.forEach(entry -> doc.put(checkFieldExists(doc, entry.key), entry.value));
+            }
+        }
+
+        private void modifyFields(Object value, String[] fieldNodes, int length) {
+            if (value instanceof List) {
+                for (Object item : (List) value) {
+                    if (item instanceof Document) {
+                        modifyFields((Document) item, fieldNodes, length);
+                    }
+                }
+            }
+        }
+
+        /**
+         * Excludes numeric items from a given array.
+         *
+         * <p>
+         * If the array has consecutive numeric items, like {@code 'a.0.0.b'}, then the numeric items aren't excluded.
+         * It is necessary because the modification of fields inside arrays of arrays isn't supported.
```

Comment:
The part between the dot characters describes a field only at one level of a document, and the document field can have a single value or array value.

Let's consider the following valid JSON:
```
{
  "numbers": [
    [
      {
        "value": 1
      },
      {
        "value": 2
      },
      {
        "value": 3
      }
    ]
  ]
}
```
If we use the following field:
```
numbers.value
```
then it can't be used to exclude the fields because the `value` field is at the 3 level of the document, not 2 level as we expect in the expression.